core/src/jsMain/kotlin/dev/fritz2/remote/http.kt

```diff
@@ -6,229 +6,306 @@ import org.w3c.fetch.*
 import org.w3c.files.Blob
 import org.w3c.xhr.FormData
 import kotlinx.browser.window as browserWindow
+import org.w3c.fetch.Response as FetchResponse
 
 
+/**
+ * creates a new [Request]
+ *
+ * @param baseUrl the common base of all urls that you want to call using the template
+ */
+fun http(baseUrl: String = "") = Request(url = baseUrl)
+
 /**
  * [Exception] type for handling http exceptions
  *
  * @property statusCode the http response status code
  * @property body the body of the error-response
  */
-class FetchException(val statusCode: Short, val body: String, val response: Response) : Exception(
+class FetchException(val statusCode: Int, val body: String, val response: Response) : Exception(
     "code=$statusCode, url=${response.url}, body=$body"
 )
 
 /**
- * creates a new [Request]
- *
- * @param baseUrl the common base of all urls that you want to call using the template
+ * Represents the common fields and attributes of an HTTP response.
+ * It contains also the original [Request] which was made to get this [Response].
  */
-fun http(baseUrl: String = "") = Request(baseUrl = baseUrl)
+open class Response(
```

Comment:
The sequence of appearance has changed. I am not sure whether this is intentional ot not. Imho the sequence in its current form does not follow any paradigm. 

If we want to present the base concepts first and subsequently add dependent concepts, the order would be:
- Response
- FetchException
- Request
- Interceptor
- fun http